db/schema.go

```diff
@@ -155,48 +156,190 @@ func createTriggerFunc(tx *pgx.Tx) error {
 	return err
 }
 
-func getTablesToRegister(conn *pgx.Conn, schemas []string, excludeTables []string) ([]string, error) {
-	exclude := make(map[string]struct{})
-	// TODO: support patterns
-	for _, t := range excludeTables {
-		exclude[t] = struct{}{}
-	}
+// GenerateTablesList using the includes and excludes list. If no tables are specified in the includes list,
+// obtain the complete list from Postgres using the supplied schemas. If any of the included tables are listed
+// as excluded, remove them from the list.
+func GenerateTablesList(conn *pgx.Conn, schemas, includeTables, excludeTables []string) ([]Table, error) {
+	tableRegister := make(map[string]bool)
 
-	var tables []string
-	for _, schema := range schemas {
-		rows, err := conn.Query(`
-		SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = $1`, schema,
-		)
-		if err != nil {
-			return nil, err
+	if len(includeTables) > 0 {
+		for _, table := range includeTables {
+			tableRegister[table] = true
 		}
-
-		for rows.Next() {
-			var t string
-			err = rows.Scan(&t)
+	} else {
+		for _, schema := range schemas {
+			rows, err := conn.Query(`
+		SELECT tablename FROM pg_catalog.pg_tables WHERE schemaname = $1`, schema,
+			)
 			if err != nil {
-				return nil, nil
+				return nil, err
 			}
 
-			if _, ok := exclude[t]; !ok {
-				tables = append(tables, fmt.Sprintf("%s.%s", schema, t))
+			for rows.Next() {
+				var t string
+				err = rows.Scan(&t)
+				if err != nil {
+					return nil, err
+				}
+
+				table := fmt.Sprintf("%s.%s", schema, t)
+				tableRegister[table] = true
 			}
 		}
 	}
 
+	for _, table := range excludeTables {
+		if _, ok := tableRegister[table]; ok {
+			tableRegister[table] = false
+		}
+	}
+
+	tables := make([]Table, 0)
+	for tableName, include := range tableRegister {
+		if include {
```

Comment:
nit: an early return (or break in this case) would reduce nesting and increase clarity